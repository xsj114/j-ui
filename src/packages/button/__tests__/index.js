import { mount } from '@vue/test-utils'
import JButton from '../index'


describe('JButton', () => {

    describe('create', () => {
        
        test('create', () => {
            const wrapper =  mount(JButton)
            expect(wrapper.html()).toBe(`<button class="j_button"></button>`)
            expect(wrapper).toMatchSnapshot()
        })

        test('create text', () => {
            const wrapper =  mount(JButton, {
                scopedSlots: {
                    default () {
                        return '添加'
                    }
                }
            })
            const buttonContent = wrapper.find('.j_button_content')
            expect(buttonContent.html()).toBe(`<span class="j_button_content">添加</span>`)
            expect(wrapper).toMatchSnapshot()
        })

        test('create icon', () => {
            const wrapper = mount(JButton,{
                scopedSlots: {
                    default: '<i class="j-icon_right_arrow"></i>'
                }
            })
            const buttonContent = wrapper.find('.j_button_content')
            const slotDefault = buttonContent.find('.j-icon_right_arrow')
            expect(slotDefault.exists()).toBe(true)
            expect(wrapper).toMatchSnapshot()
        })


        test('click', async () => {
            const onClick = jest.fn()
            const wrapper = mount(JButton, {
                scopedSlots: {
                    default () {
                        return '添加'
                    }
                },
                listeners: {
                    click: onClick
                }
            })
            await wrapper.trigger('click')
            expect(onClick).toHaveBeenCalled()
            expect(onClick).toHaveBeenCalledTimes(1)
            expect(wrapper).toMatchSnapshot()
        })

    })

    describe('size', () => {

        test('small size', () => {

            const wrapper = mount(JButton,{
                propsData: {
                    size: 'small'
                }
            })
            expect(wrapper.classes()).toContain('j_button_small')
            expect(wrapper).toMatchSnapshot()

        })

        test('large size', () => {

            const wrapper = mount(JButton,{
                propsData: {
                    size: 'large'
                }
            })
            expect(wrapper.classes()).toContain('j_button_large')
            expect(wrapper).toMatchSnapshot()

        })
    })

    describe('round', () => {

        test('is round', () => {

            const wrapper = mount(JButton, {
                propsData: {
                    round: true
                }
            })
            expect(wrapper.classes()).toContain('j_button_round')
            expect(wrapper).toMatchSnapshot()

        })

    })


    describe('border', () => {

        test('is border', () => {
            const wrapper = mount(JButton, {
                propsData: {
                    border: true
                }
            })
            expect(wrapper.classes()).toContain('j_button_border')
            expect(wrapper).toMatchSnapshot()
        })
    })


    describe('disabled', () => {

        test('is disabled', () => {
            const wrapper = mount(JButton, {
                propsData: {
                    disabled: true
                }
            })
            expect(wrapper.attributes('disabled')).toBe('disabled')
            expect(wrapper).toMatchSnapshot()
        })
    })

})
