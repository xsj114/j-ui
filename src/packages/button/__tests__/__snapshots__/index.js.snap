// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JButton border is border 1`] = `<button class="j_button j_button_border"></button>`;

exports[`JButton create click 1`] = `<button class="j_button"><span class="j_button_content">添加</span></button>`;

exports[`JButton create create 1`] = `<button class="j_button"></button>`;

exports[`JButton create create icon 1`] = `<button class="j_button"><span class="j_button_content"><i class="j-icon_right_arrow"></i></span></button>`;

exports[`JButton create create text 1`] = `<button class="j_button"><span class="j_button_content">添加</span></button>`;

exports[`JButton disabled is disabled 1`] = `<button disabled="disabled" class="j_button"></button>`;

exports[`JButton round is round 1`] = `<button class="j_button j_button_round"></button>`;

exports[`JButton size large size 1`] = `<button class="j_button j_button_large"></button>`;

exports[`JButton size small size 1`] = `<button class="j_button j_button_small"></button>`;
